using-kubecost/navigating-the-kubecost-ui/savings/cluster-right-sizing-recommendations.md

```diff
@@ -81,3 +86,1150 @@ If you are using Persistent Volumes (PVs) with AWS's Elastic Block Store (EBS) C
 Kubecost mitigates this problem by ensuring continuous cluster right-sizing creates at least one node per AZ by forcing NodeGroups to have a node count greater than or equal to the number of AZs of the EKS cluster. This will also prevent you from setting a minimum node count for your recommendation below the number of AZs for your cluster. If the EBS CSI continues to be problematic, you can consider switching your CSI to services like Elastic File System (EFS) or FSx for Lustre.
 
 Using Cluster Autoscaler on AWS may result in a similar error. See more [here](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md#common-notes-and-gotchas).
+
+## Instance Selection 
+
+Cluster right-sizing recommendations are generated by simulating placing your cluster workload requirements on multiple different instance type configurations and choosing the most resource- and cost-optimal option. Kubecost queries public pricing APIs to build and maintain a ‘general pool’ of instance type options for every supported cloud service provider.
+However, your organization may limit the instance types that can be provisioned, and not all nodes in our 'general pool' may be acceptable in your use case.
+
+### Configuration
+You can supply your desired list of node types in an allow list; this way, Kubecost will only evaluate the specified instance types when generating recommendations. We support allow lists for AWS, GCP and Azure instance types.
+Each of the allow lists can be configured by setting the related [Helm chart parameters](https://github.com/kubecost/cost-analyzer-helm-chart/blob/d5144c1c5354e2978b56194f10d3a87cd545a100/cost-analyzer/values.yaml#L3450-L3453):
+
+```yaml
+kubecostProductConfigs:
+  ...
+  savingsRecommendationsAllowLists:
+    AWS:
+      - t3.2xlarge
+      - t3.large
+      - t3.medium
+      - t3.micro
+      - t3.nano
+      - t3.small
+      - t3.xlarge
+      - ...
+    GCP: 
+      - e2-standard-2
+      - ...
+    Azure:
+      - B1ms
+      - ...
+```
+
+### Supported instance types
+
+Below are the complete lists of supported instance types that are currently available for each of the supported cloud service providers:
```

Comment:
Woah these lists are insanely long. @srpomeroy or @chipzoller do you have any guidance on how we can present this better?

Reply:
I think you may be able to wrap all the instance types for each provider in a dropdown. Something like this I think?

```
<details>
<summary> AWS </summary>

- a1.2xlarge
- a1.4xlarge
- a1.large
- a1.medium
...

</details>
```

Reply:
I believe each PR has an associated "Gitbook Preview". This will show you what it will look like live on the docs site!

Reply:
I don't see a native way to collapse or add vertical scrolling. I opened a support request to see if they can advise.
[Play with it in GitBook](https://app.gitbook.com/o/MQuX6uFwV0j7vIHtR15E/s/qsPmTvJxhcJ8DiSO6JHf/)

[instancetypes.md](https://github.com/user-attachments/files/16883443/instancetypes.md)

<img width="871" alt="image" src="https://github.com/user-attachments/assets/ff430b9c-eb93-4040-b5fa-a62f998a2291">

Reply:
> Are these instance types hard-coded in the product? Any chance they are in OpenCost and we can just reference the code?
Sean - no, unfortunately the node types are not in opencost, they're in KCM

Reply:
> I don't see a native way to collapse or add vertical scrolling. I opened a support request to see if they can advise. 

@srpomeroy That's great, thank you for this!

Reply:
Alternatively, since this is a helm config, we could publish 3 files (one per provider) in the **cost-analyzer-helm-chart** repo as assets/resources and then we could reference them by their URLs on this public docs page. What do you all think about this? Might be somewhat simpler

Reply:
I prefer that method and we just keep the abridged examples here.

Reply:
I like that idea as well!

Reply:
I've just raised a helm chart PR here: https://github.com/kubecost/cost-analyzer-helm-chart/pull/3645 and modified the docs to reference those files instead.